.github/workflows/workbench.yml

```
@@ -0,0 +1,116 @@
+name: Workbench
+
+on:
+  workflow_call:
+    inputs:
+      type:
+        required: true
+        type: string
+      game_appid:
+        required: true
+        type: string
+      tools_appid:
+        required: true
+        type: string
+
+env:
+  WORKBENCH_COMMON_ARGS: -enableWARP -exitAfterInit -noThrow -noSound -scriptAuthorizeAll -silentCrashReport -VMErrorMode fatal
+
+jobs:
+  workbench:
+    name: Workbench
+    runs-on: windows-2022
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          path: ACE-Anvil
+
+      - name: Checkout pull request
+        uses: actions/checkout@v4
+        if: ${{ github.event_name == 'pull_request_target' }}
+        with:
+          path: ACE-Anvil-PullRequest
+          ref: refs/pull/${{ github.event.number }}/merge
+
+      - name: Replace addons with pull request addons
+        if: ${{ github.event_name == 'pull_request_target' }}
+        run: |
+          rm -r ACE-Anvil\addons\
+          xcopy /e /h /q ACE-Anvil-PullRequest\addons ACE-Anvil\addons\
+
+      - name: Install DepotDownloader
+        run: |
+          $url = 'https://github.com/SteamRE/DepotDownloader/releases/download/DepotDownloader_3.0.0/DepotDownloader-windows-x64.zip'
+          Invoke-RestMethod -Uri $url -OutFile DepotDownloader.zip
+          Expand-Archive DepotDownloader.zip -DestinationPath c:\DepotDownloader -Force
+          Remove-Item DepotDownloader.zip
+
+      - name: Install Arma Reforger
+        run: c:\DepotDownloader\DepotDownloader.exe -dir c:\reforger -app ${{ inputs.game_appid }} -username "${{ secrets.STEAM_USERNAME }}" -password "${{ secrets.STEAM_PASSWORD }}"
+
+      - name: Install Arma Reforger Tools
+        run: c:\DepotDownloader\DepotDownloader.exe -dir C:\tools -app ${{ inputs.tools_appid }} -username "${{ secrets.STEAM_USERNAME }}" -password "${{ secrets.STEAM_PASSWORD }}"
+
+      - name: Steam DLL
+        run: |
+          $url = '${{ secrets.STEAM_API64_DLL_URL }}'
+          Invoke-RestMethod -Uri $url -OutFile c:/tools/Workbench/steam_api64.dll
+
+      - name: Validate Scripts
+        shell: pwsh
+        run: |
+          $AddonsDir = "C:/reforger/addons/,${{ github.workspace }}/ACE-Anvil/addons/"
+          $AddonGuid = "60C4E0B49618CC62"
+          $proc = Start-Process -PassThru -Wait -FilePath "c:/tools/Workbench/ArmaReforgerWorkbenchSteamDiag.exe" -ArgumentList "$env:WORKBENCH_COMMON_ARGS -wbModule=ScriptEditor -addonsDir ""$AddonsDir"" -addons ""$AddonGuid"" -validate"
```

From my tests it looks like `-wbSilent` is needed for non 0 status code if scripts are wrong.
```suggestion
          $proc = Start-Process -PassThru -Wait -FilePath "c:/tools/Workbench/ArmaReforgerWorkbenchSteamDiag.exe" -ArgumentList "-wbSilent $env:WORKBENCH_COMMON_ARGS -wbModule=ScriptEditor -addonsDir ""$AddonsDir"" -addons ""$AddonGuid"" -validate"
```